lib/core/osal/netdev_funcs.c

```diff
@@ -236,11 +238,41 @@ netdev_get_offloads(const char *ifname, struct offloads *off)
     return ret;
 }
 
+static void
+netdev_get_channels_from_sysfs(const char *if_name, uint32_t *rxq, uint32_t *txq)
+{
+    char buf[PATH_MAX];
+    struct dirent *entry;
+    DIR *dir;
+    int ret;
+
+    *rxq = *txq = 0;
+
+    ret = snprintf(buf, PATH_MAX, "/sys/class/net/%s/queues/", if_name);
```

Comment:
I think snprintf() returns the number of bytes in the string, which means the 'if' below would return without doing anything. I think the test should be if(ret >= PATH_MAX) CNE_RET("invalid path length\n");

Reply:
yes I will update